pkg/oauthserver/oauth/registry/userconversion.go

```diff
@@ -5,59 +5,64 @@ import (
 
 	kuser "k8s.io/apiserver/pkg/authentication/user"
 
-	oapi "github.com/openshift/api/oauth/v1"
+	oauthapi "github.com/openshift/api/oauth/v1"
 )
 
-type UserConversion struct{}
+type UserConversion interface {
+	ConvertToAuthorizeToken(interface{}, *oauthapi.OAuthAuthorizeToken) error
+	ConvertToAccessToken(interface{}, *oauthapi.OAuthAccessToken) error
+	ConvertFromAuthorizeToken(*oauthapi.OAuthAuthorizeToken) (kuser.Info, error)
+	ConvertFromAccessToken(*oauthapi.OAuthAccessToken) (kuser.Info, error)
+}
+
+type userConversion struct{}
 
 // NewUserConversion creates an object that can convert the user.Info object to and from
 // an oauth access/authorize token object.
-func NewUserConversion() *UserConversion {
-	return &UserConversion{}
+func NewUserConversion() UserConversion {
+	return &userConversion{}
 }
 
-func (s *UserConversion) ConvertToAuthorizeToken(user interface{}, token *oapi.OAuthAuthorizeToken) error {
-	info, ok := user.(kuser.Info)
-	if !ok {
-		return errors.New("did not receive user.Info")
-	}
-	token.UserName = info.GetName()
-	if token.UserName == "" {
-		return errors.New("user name is empty")
-	}
-	token.UserUID = info.GetUID()
-	return nil
+func (s *userConversion) ConvertToAuthorizeToken(user interface{}, token *oauthapi.OAuthAuthorizeToken) (err error) {
```

Comment:
I thought we avoided named returns overall. (not a big deal for a small function)

Reply:
I was being a bit too cute in an attempt to save a line.  I dropped the named return here but kept it in the later functions to make it easier to distinguish what each string is.  We avoid bare returns, which are only possible if you use named returns.  In general using named returns to initialize variables is hard to read, but they work really nicely in interfaces since that only ever serves as documentation.